src/serve.rs

```diff
@@ -42,19 +51,51 @@ impl ServeSystem {
             }
         }
 
+        if self.cfg.autoreload {
+            spawn(Self::build_notify_task(self.build_done_rx, websockets));
+        }
+
         server_handle.await;
         watch_handle.await;
         Ok(())
     }
 
+    async fn build_notify_task(mut build_done_rx: Receiver<()>, websockets: WebSockets) {
+        while let Some(_) = build_done_rx.next().await {
+            let mut ws_guard = websockets.lock().await;
+            let mut to_remove = vec![];
+            for (idx, ws) in ws_guard.iter_mut().enumerate() {
+                if ws.send_string(r#"{"reload": true}"#.to_string()).await.is_err() {
+                    to_remove.push(idx);
+                }
+            }
+            for idx in to_remove.into_iter().rev() {
+                ws_guard.remove(idx);
+            }
+        }
+    }
```

Comment:
Once we cut over to tokio, we can simply clone the broadcaster held in the server state, spawn a task which listens for the build completion signals & also listen for the WebSocket to close. That would simplify the setup here quite a lot, and would remove the need for any synchronization (mutex).

For now, solid work `:)`!